### ISSUE TITLE
left-over files from deleting a frozen container

### ISSUE THREAD

**Issue by yogo1212** [containers/podman#22016]
### Issue Description

Every now and then, I encounter an container that in a limbo state.
It's neither running nor being listed with `podman container ls --all` but it's name can't be re-used either.
Typically, the podman process was interrupted using ctrl-c before but then failed to terminate correctly.
It seems to happen most with the postgres container.

```
time="2024-03-11T21:06:04Z" level=warning msg="Unmounting container \"postgres\" while attempting to delete storage: unmounting \"/home/postgres/.local/share/containers/storage/overlay/6e06361057c3ef4ad4537ab6c9a19647817ff7e0e387f5a3da1589fe87349103/merged\": invalid argument"
```

A `sudo rm -rf /home/postgres/.local/share/containers/storage/overlay/6e06361057c3ef4ad4537ab6c9a19647817ff7e0e387f5a3da1589fe87349103` appears to suffice for recovery.

Maybe related to https://github.com/containers/podman/issues/12316?

### Steps to reproduce the issue

- run `podman run --rm --name "postgres" -p 5432:5432 --userns keep-id --hostuser=postgres --env-file /home/postgres/postgres.env -v "some/dir/for/postgres:/var/lib/postgresql/data" docker.io/library/postgres:14-alpine`
- connect an active client
- send sigint (ctrl-c) to podman
- sometimes, it doesn't react
  - great, the reproduction is working
  - `podman rm --ignore -f -t 10 "postgres"`
- try step 1 again
  - if in the error state, `podman run` won't work (either because of the name not being available or due to failed cleanup, see further down)
  - in that case, stop :wink: 

### Describe the results you received

There are still files related to the removed container, like:

```
$ cat overlay-containers/volatile-containers.json 
[{"id":"564aef3136169972e10c828f2971584dcc8a0de584a7237265585dbb85e4978e","names":["postgres"],"image":"54ae7e6e096bdd77229713bb3a17983e82b7def419a0a78ef4b1102478515c0b","layer":"6e06361057c3ef4ad4537ab6c9a19647817ff7e0e387f5a3da1589fe87349103","metadata":"{\"image-name\":\"docker.io/library/postgres:14-alpine\",\"image-id\":\"54ae7e6e096bdd77229713bb3a17983e82b7def419a0a78ef4b1102478515c0b\",\"name\":\"postgres\",\"created-at\":1710172119}","created":"2024-03-11T15:48:39.069743504Z","uidmap":[{"container_id":0,"host_id":1,"size":1003},{"container_id":1003,"host_id":0,"size":1},{"container_id":1004,"host_id":1004,"size":64533}],"gidmap":[{"container_id":0,"host_id":1,"size":1003},{"container_id":1003,"host_id":0,"size":1},{"container_id":1004,"host_id":1004,"size":64533}],"flags":{"MountLabel":"","ProcessLabel":"","Volatile":true}}]
```

Some of those files belong to the first mapped user/group (`/etc/sub{u,g}id`) and can't be deleted without root:

```
drwx------ 4   296608   296608 4096 Mar 11 20:35 overlay-containers/564aef3136169972e10c828f2971584dcc8a0de584a7237265585dbb85e4978e/userdata
..
drwx------  5   296608   296608 4096 Mar 11 15:48 overlay/6e06361057c3ef4ad4537ab6c9a19647817ff7e0e387f5a3da1589fe87349103
```

### Describe the results you expected

All files relating to containers should be gone.

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.31.3
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.8-1.el9.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.8, commit: cebaba63f66de0e92cdc7e2a59f39c9208281158'
  cpuUtilization:
    idlePercent: 92.4
    systemPercent: 1.96
    userPercent: 5.65
  cpus: 2
  databaseBackend: boltdb
  distribution:
    distribution: '"rocky"'
    version: "9.3"
  eventLogger: journald
  freeLocks: 2047
  hostname: nope.com
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
  kernel: 5.14.0-362.18.1.el9_3.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 372641792
  memTotal: 3724410880
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.7.0-1.el9.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.7.0
    package: netavark-1.7.0-2.el9_3.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.7.0
  ociRuntime:
    name: crun
    package: crun-1.8.7-1.el9.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.8.7
      commit: 53a9996ce82d1ee818349bdcc64797a1fa0433c4
      rundir: /run/user/1001/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  pasta:
    executable: ""
    package: ""
    version: ""
  remoteSocket:
    path: /run/user/1001/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.1-1.el9.x86_64
    version: |-
      slirp4netns version 1.2.1
      commit: 09e31e92fa3d2a1d3ca261adaeb012c8d75a8194
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 3822055424
  swapTotal: 4294963200
  uptime: 4h 58m 48.00s (Approximately 0.17 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/xxx/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/xxx/.local/share/containers/storage
  graphRootAllocated: 40165670912
  graphRootUsed: 21927145472
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/1001/containers
  transientStore: false
  volumePath: /home/xxx/.local/share/containers/storage/volumes
version:
  APIVersion: 4.6.1
  Built: 1709719721
  BuiltTime: Wed Mar  6 10:08:41 2024
  GitCommit: ""
  GoVersion: go1.20.12
  Os: linux
  OsArch: linux/amd64
  Version: 4.6.1
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

Rocky Linux 9.3

**Comment by giuseppe** [comment]
there were some improvements in this area since 4.6.1.  Please try with a newer version

**Comment by Luap99** [comment]
Yes I am very sure this was fixed in 4.7
Duplicate of #19913
